robot_ws/src/martian_detector/nodes/object_detector

```diff
@@ -13,92 +13,95 @@
  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 """
-
 import rospy
 import boto3
 import time
 
 from std_msgs.msg import String
 from sensor_msgs.msg import CompressedImage, Joy
 
-CAMERA_TOPIC = rospy.get_param('~/object_detector/camera_topic', '/camera/image_raw/compressed')
-AWS_REGION = rospy.get_param('~/object_detector/aws_region', 'us-west-2')
-AUTO_CAPTURE = rospy.get_param('~/object_detector/auto_capture', True)
-AUTO_CAPTURE_RATE = 1 # When using auto capture, take a photo every second
-IMAGE_CAPTURE_WAIT = 3 # At least 3 seconds between image captures to not spam rekognition
+
+MIN_CAPTURE_PAUSE = 3 # Min seconds between image captures, no need to spam services here
+MIN_CONFIDENCE = 50  # 0 to 100
+MAX_LABEL_COUNT = 10
+
 
 class ObjectDetector():
 
     def __init__(self):
+        rospy.loginfo('(ObjectDetector): initializing')
         self.pub = rospy.Publisher('/detected_objects', String, queue_size=1)
-        self.rekog = boto3.client('rekognition', AWS_REGION)
-        self.joysub = rospy.Subscriber("/joy", Joy, self.joystick_input)
-        self.last_image_time = 0
+        self.joysub = rospy.Subscriber("/joy", Joy, self.joystick_input)  # Kept this to show how to hook joystick if one is available
+        
+        self.aws_region = rospy.get_param('~/object_detector/aws_region', 'us-west-2')
```

Comment:
Same as the nested default comment above!

Reply:
will remove